File: code-defense/player/runner.cpp
#include <vector>
#include <string>
#include <deque>
#include <chrono>
#include <cstdio>
#include <cstdlib>
#include <algorithm>
using namespace std;

int watchTheFlow(vector<int>& current, int target);

int main(int argc, char* argv[]) {
    bool test_only = (argc > 1 && string(argv[1]) == "--test");
    bool all_correct = true;
    int ops = 0;

    auto start = chrono::high_resolution_clock::now();


    auto end = chrono::high_resolution_clock::now();
    int ms = (int)chrono::duration_cast<chrono::milliseconds>(end - start).count();

    printf("%d %d\n", ms, ops);
    return all_correct ? 0 : 1;
}

File: code-defense/player/runner.js
const { walkThread } = require('./solution');
const testOnly = process.argv.includes('--test');
let allCorrect = true;
let ops = 0;
const start = process.hrtime.bigint();

{
    let t = [3, 1, 4, 1, 5, 1];
    if (walkThread(t, 1) !== 3) allCorrect = false;
}
{
    let t = [7, 7, 7];
    if (walkThread(t, 7) !== 3) allCorrect = false;
}
{
    let t = [1, 2, 3];
    if (walkThread(t, 5) !== 0) allCorrect = false;
}
{
    let t = [42];
    if (walkThread(t, 42) !== 1) allCorrect = false;
}
if (!testOnly) {
    let n = 100000;
    let t = new Array(n);
    for (let i = 0; i < n; i++) t[i] = i % 10;
    if (walkThread(t, 0) !== Math.floor(n / 10)) allCorrect = false;
    ops = n;
} else { ops = 13; }

const end = process.hrtime.bigint();
const ms = Number((end - start) / 1000000n);
process.stdout.write(`${ms} ${ops}\n`);
process.exit(allCorrect ? 0 : 1);
